homeassistant/components/fritzbox_callmonitor/config_flow.py

```diff
@@ -0,0 +1,255 @@
+        self._prefixes = None
+        self._serial_number = None
+
+    def _get_config_entry(self):
+        """Create and return an config entry."""
+        return self.async_create_entry(
+            title=self._phonebook_name,
+            data={
+                CONF_HOST: self._host,
+                CONF_PORT: self._port,
+                CONF_USERNAME: self._username,
+                CONF_PASSWORD: self._password,
+                CONF_PHONEBOOK: self._phonebook_id,
+                CONF_PREFIXES: self._prefixes,
+                SERIAL_NUMBER: self._serial_number,
+            },
+        )
+
+    def _try_connect(self):
+        """Try to connect and check auth."""
+        self._phonebook = FritzBoxPhonebook(
+            host=self._host,
+            username=self._username,
+            password=self._password,
+            phonebook_id=self._phonebook_id,
+            prefixes=self._prefixes,
+        )
+
+        try:
+            self._phonebook.init_phonebook()
+            self._phonebook_ids = self._phonebook.get_phonebook_ids()
+
+            fritz_connection = FritzConnection(
+                address=self._host, user=self._username, password=self._password
+            )
+            device_info = fritz_connection.call_action(
+                FRITZ_SERVICE_DEVICE_INFO, FRITZ_ACTION_GET_INFO
+            )
+            self._serial_number = device_info[FRITZ_ATTR_SERIAL_NUMBER]
+
+            return RESULT_SUCCESS
+        except RequestsConnectionError:
+            return RESULT_NO_DEVIES_FOUND
+        except FritzSecurityError:
+            return RESULT_INSUFFICIENT_PERMISSIONS
+        except FritzConnectionException:
+            return RESULT_INVALID_AUTH
+
+    def _is_already_configured(self, host, phonebook_id):
+        """Check if an entity with the same host and phonebook_id is already configured."""
+        for config_entry in self.hass.config_entries.async_entries(DOMAIN):
+            if (
+                config_entry.data[CONF_HOST] == host
+                and config_entry.data[CONF_PHONEBOOK] == phonebook_id
+            ):
+                return True
+
+    async def _get_name_of_phonebook(self, phonebook_id):
+        """Return name of phonebook for given phonebook_id."""
+        phonebook_info = await self.hass.async_add_executor_job(
+            self._phonebook.fph.phonebook_info, phonebook_id
+        )
+        return phonebook_info[FRITZ_ATTR_NAME]
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return FritzBoxCallMonitorOptionsFlowHandler(config_entry)
+
+    async def async_step_import(self, user_input=None):
+        """Handle configuration by yaml file."""
+        return await self.async_step_user(user_input)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
```

Comment:
You can save on indentation by doing:

```python
if user_input is None:
    return self.async_show_form(
        step_id="user", data_schema=DATA_SCHEMA_USER, errors={}
    )
```

Reply:
You are right, but I like it like this because I am not writing the following return statement twice, once with an empty error dict and one with a filled dict.
```python
if user_input is None:
    return self.async_show_form(
        step_id="user", data_schema=DATA_SCHEMA_USER, errors={}
    )
...
errors["base"] = result # "invalid_auth"
return self.async_show_form(
    step_id="user", data_schema=DATA_SCHEMA_USER, errors=errors
)
```